### ISSUE TITLE
SC2116 does not apply when variables are expanded indirectly from an array

### ISSUE THREAD

**Issue by AleixMT** [koalaman/shellcheck#2637]
#### For bugs
- [x] SC2116
- [ ] I am using IntelliJ IDEA ultimate 2021.2 and the plugin for [shell scripts](https://plugins.jetbrains.com/plugin/13122-shell-script), which includes Shellcheck. I am really sorry but I was not able to find out which version I have; but I am sure that I am not on legacy versions since this is a rule that has not been changed in a long time. I have been seeing this shellcheck complain for years.
- [x] The rule's wiki page does not already cover this! It mentions that it can be because of the side effect of `echo` or expansion but it does not say explicitly that indirect expansion of variables that reference arrays with more than one element or a single element that contains spaces is an exception of this rule. 
- [x] Yes, I tried it! Here is the screenshot as proof:
- 
![image](https://user-images.githubusercontent.com/23342150/207362440-9c108b82-76ba-4454-9370-f666f274ba30.png)

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

generic()
{
  CURRENT_INSTALLATION_KEYNAME=baseVariable
  
  baseVariable_pipinstallations=("python-dotenv" "two_or_more")  # With this we would have an error
  baseVariable_pythoncommands=("one_no_error")   # This does not fail because the expanded array has a single element
  
  local -r pipinstallations="${CURRENT_INSTALLATION_KEYNAME}_pipinstallations[@]"
  local -r pythoncommands="${CURRENT_INSTALLATION_KEYNAME}_pythoncommands[@]"
  if [ -z "$(echo "${!pipinstallations}")" ] && [ -z "$(echo "${!pythoncommands}")" ]; then  # This line suggests a useless echo, but if we remove the echo the code will fail with script.sh: line 1434: [: too many arguments or will fail with script.sh: line 1434: [: pythondotenv: binary operator expected
    return
  fi
}
```

#### Here's what shellcheck currently says:
Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'

#### Here's what I wanted or expected to see:
Nothing, it is an exception of this rule. When using `echo` around indirect expanded variables that are an array with more than one element, shellcheck should not show this error because when you actually remove the error the code stops working as expected. The suggested code will make my new code look like this:
```
  if [ -z "${!pipinstallations}" ] && [ -z "${!pythoncommands}" ]; then
    return
  fi
```sh

But it will fail with `script.sh: line 1434: [: too many arguments` or will fail with `script.sh: line 1434: [: pythondotenv: binary operator expected` if the array `pipinstallations` or the array `pythoncommands` contains more than one element, or it has one element with spaces in it. 

I hope that this is actually a request that makes sense, since from my point of view this shellcheck is wrong in the special circumstances described.
